glean-core/android/src/main/java/mozilla/telemetry/glean/rust/LibGleanFFI.kt

```diff
@@ -62,16 +63,21 @@ internal fun Pointer.getRustString(): String {
     return this.getString(0, "utf8")
 }
 
-@Suppress("TooManyFunctions")
-internal interface LibGleanFFI : Library {
-    companion object {
-        private val JNA_LIBRARY_NAME = "glean_ffi"
-
-        internal var INSTANCE: LibGleanFFI = try {
-            val lib = Native.load(JNA_LIBRARY_NAME, LibGleanFFI::class.java) as LibGleanFFI
-            lib.glean_enable_logging()
-            lib
+/**
+ * Load the provided library by name.
+ * If it can't be loaded try loading `libxul.so` instead.
+ * If that also fails, give up and raise an exception.
+ */
+internal fun loadIndirect(libraryName: String): LibGleanFFI {
+    val lib = try {
+        Native.load(libraryName, LibGleanFFI::class.java) as LibGleanFFI
+    } catch (e: UnsatisfiedLinkError) {
+        Log.i("glean/loadIndirect", "Failed to load $libraryName")
+        try {
+            Log.i("glean/loadIndirect", "Trying to load libxul directly")
+            Native.load("xul", LibGleanFFI::class.java) as LibGleanFFI
```

Comment:
Eventually we'll probably want to prefer libxul's Glean over libgleanffi for a couple reasons
1) Performance micro-optimization that we don't need to fail to load a lib on every startup in the "usual" case
2) In the event that we do have both, we probably want to prefer the libxul-present one since that's the one libxul itself will be using.

Modulo us shipping this to Android first and m-c second, of course. If we ship this code to Android first, it'll succeed at finding a libxul and not fall through to loading libgleanffi. (It should fail at finding symbols to satisfy LibGleanFFI... but I could see us not wanting to rely on that.)